### PR TITLE
Add documentation for credentials to website

### PR DESCRIPTION
**Pull request by mgaffney**

### CODE REVIEW

File: website/content/docs/concepts/domain-model/credential-stores.mdx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  capabilities = ["update"]
+}
+```
+
+The above [`boundary-controller` policy](/data/vault/boundary-controller-policy.hcl) is
+available for download. Below is an example of writing this policy to Vault:
+
+```shell-session
+# Download the policy
+$ curl https://boundaryproject.io/data/vault/boundary-controller-policy.hcl -O -s -L
```
Comment (malnick) [inline website/content/docs/concepts/domain-model/credential-stores.mdx:154]: Love the fact that we give the policy here as a downloadable link :)

**Review by malnick — APPROVED** [review]
Looks great!

**Review by louisruch — APPROVED** [review]
Looks great!

File: website/content/docs/concepts/domain-model/credential-stores.mdx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+of differing types and differing access levels.
+It belongs to a [project][] and supports the
+[principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)
+by providing mechanisms to limit the [credentials][] it can access
+to the minimum necessary for the project it is in.
+A credential store can also contain [credential libraries][].
+
+## Attributes
+
+A credential store has the following configurable attributes:
```
Comment (louisruch) [inline website/content/docs/concepts/domain-model/credential-stores.mdx:21]: ScopeId?

File: website/content/docs/concepts/domain-model/credential-libraries.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+# Credential Libraries
+
+A credential library is a resource
+that provides [credentials][]
+of the same type and same access level
+from a single [credential store][].
+
+## Attributes
+
+A credential library has the following configurable attributes:
```
Comment (louisruch) [inline website/content/docs/concepts/domain-model/credential-libraries.mdx:17]: CredStoreId?

File: website/content/docs/concepts/domain-model/credential-stores.mdx (hunk 1 above)
Comment (mgaffney) [reply website/content/docs/concepts/domain-model/credential-stores.mdx:154]: ty!

File: website/content/docs/concepts/domain-model/credential-stores.mdx (hunk 2 above)
Comment (mgaffney) [reply website/content/docs/concepts/domain-model/credential-stores.mdx:21]: We don't include parent IDs in the other domain docs. Those are documented in the API docs.

File: website/content/docs/concepts/domain-model/credential-libraries.mdx (hunk 3 above)
Comment (mgaffney) [reply website/content/docs/concepts/domain-model/credential-libraries.mdx:17]: Same answer

**Pull request merged**